### ISSUE TITLE
can't open file 'setup.py': [Errno 2] No such file or directory

### ISSUE THREAD

**Issue by matrixguy181** [pytorch/pytorch#9578]
So I'm trying to install PyTorch from source.  But when I'm trying to run the final command,
`MACOSX_DEPLOYMENT_TARGET=10.9 CC=clang CXX=clang++ python setup.py install` 
my terminal throws this
`python: can't open file 'setup.py': [Errno 2] No such file or directory`

So how can I get rid of this and install PyTorch?

INFO :
OS : macOS 10.13.6
CUDA version : 9.2
cuDNN : 9.2
xcode CLT ( macOS 10.12 version 8.2)
Anaconda3 is installed. 
`clang --version`
`Apple LLVM version 8.0.0 (clang-800.0.42.1)`
`Target: x86_64-apple-darwin17.7.0`
`Thread model: posix`
`InstalledDir: /Library/Developer/CommandLineTools/usr/bin`

**Comment by bhushan23** [comment]
are you sure you have synced correctly and you are indeed in correct directory?
also do git submodule update --init (This is not required for above error)

**Comment by matrixguy181** [comment]
I'm running this command from my home directory. Which directory am I supposed to be in?

**Comment by matrixguy181** [comment]
@bhushan23 , I'm running this command from my home directory. Which directory am I supposed to be in?

**Comment by soumith** [comment]
@matrixguy181 you need to be in the freshly cloned pytorch directory. https://github.com/pytorch/pytorch/blob/master/README.md#get-the-pytorch-source

**Comment by matrixguy181** [comment]
@soumith, As you said I ran `MACOSX_DEPLOYMENT_TARGET=10.9 CC=clang CXX=clang++ python setup.py install` in the pytorch directory and it seemed to work fine (It ran for an hour. Isn't it unusual?). But `torch.cuda.is_available()` still return `False`. I don't know what's going wrong. The log is pretty long so I'm posting last few lines of the log. 

`copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THGenerateDoubleType.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THGenerateShortType.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THGenerateIntTypes.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THGenerateLongType.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THAllocator.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THGenerateCharType.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THSize.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THMemoryFile.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THStorage.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THHalf.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THStorage.hpp -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THGenerateHalfType.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THGenerateIntType.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THVector.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THGeneral.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THLapack.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THRandom.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THLogAdd.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THGenerateFloatTypes.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/generic/THBlas.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH/generic
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/generic/THTensorMath.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH/generic
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/generic/THTensorFastGetSet.hpp -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH/generic
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/generic/THTensorConv.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH/generic
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/generic/THStorage.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH/generic
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/generic/THTensorCopy.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH/generic
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/generic/THTensorLapack.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH/generic
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/generic/THTensorRandom.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH/generic
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/generic/THVector.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH/generic
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/generic/THStorageCopy.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH/generic
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/generic/THLapack.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH/generic
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/generic/THTensor.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH/generic
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THTensor.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THMath.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THDiskFile.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/TH.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THTensorApply.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THGenerateAllTypes.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THTensor.hpp -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THGenerateByteType.h -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/include/TH/THGenerator.hpp -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib/include/TH
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/libtorch.1.dylib -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib
copying build/lib.macosx-10.9-x86_64-3.6/torch/lib/libshm.dylib -> /Users/KP/anaconda3/lib/python3.6/site-packages/torch/lib
byte-compiling /Users/KP/anaconda3/lib/python3.6/site-packages/torch/version.py to version.cpython-36.pyc
running install_egg_info
running egg_info
writing torch.egg-info/PKG-INFO
writing dependency_links to torch.egg-info/dependency_links.txt
writing top-level names to torch.egg-info/top_level.txt
reading manifest file 'torch.egg-info/SOURCES.txt'
writing manifest file 'torch.egg-info/SOURCES.txt'
removing '/Users/KP/anaconda3/lib/python3.6/site-packages/torch-0.5.0a0+4c615b1-py3.6.egg-info' (and everything under it)
Copying torch.egg-info to /Users/KP/anaconda3/lib/python3.6/site-packages/torch-0.5.0a0+4c615b1-py3.6.egg-info
running install_scripts
kp:pytorch KP$ `

**Comment by soumith** [comment]
@matrixguy181 do you have CUDA installed on your Mac, and do you have a NVIDIA GPU on your mac?

**Comment by matrixguy181** [comment]
@soumith , CUDA v9.2 is installed (the latest version). I have GeForce GT 650M.  I also downloaded cuDNN v7.1.4 (latest version) and moved the library files to cuda folder by running the commands below.

`sudo mv cudnn.h  /usr/local/cuda/include/`
`sudo mv libcudnn*  /usr/local/cuda/lib/ `

Then I downgraded xcode CLT to v8.2, set the cmake prefix path, cloned pytorch. Building also seemed to work just fine. What am I missing ?
